### ISSUE TITLE
Leftover browsers in selenium.

### ISSUE THREAD

**Issue by jaryn** [ManageIQ/integration_tests#4228]
Recently I found that our selenium server (no wharf was used) started filling-up with browsers that seems to be leftovers. And for some reason, tests started failing like so:

```
2017-02-20 03:16:00,655 [I] [UI-NAV/Server/LoggedIn]: Not here (utils/appliance/implementations/ui.py:355)
2017-02-20 03:16:00,655 [I] [UI-NAV/Server/LoginScreen]: Op is a Nop! (pre_navigate) (utils/appliance/implementations/ui.py:355)
2017-02-20 03:16:00,656 [I] [UI-NAV/Server/LoginScreen]: Checking if already here (utils/appliance/implementations/ui.py:355)
2017-02-20 03:16:00,656 [I] starting browser (utils/browser.py:324)
2017-02-20 03:16:00,665 [I] [UI-NAV/Server/LoginScreen]: Exception raised [assert <Thread(Thread-3, started daemon 140532765783808)> is None
 +  where <Thread(Thread-3, started daemon 140532765783808)> = <Thread(Thread-3, started daemon 140532765783808)>
 +    where <Thread(Thread-3, started daemon 140532765783808)> = <utils.browser.BrowserManager object at 0x3505610>._browser_renew_thread] whilst checking if already here (utils/appliance/implementations/ui.py:355)
2017-02-20 03:16:00,665 [I] [UI-NAV/Server/LoginScreen]: Not here (utils/appliance/implementations/ui.py:355)
2017-02-20 03:16:00,666 [I] starting browser (utils/browser.py:324)
2017-02-20 03:16:00,674 [E] Error setting up provider ci-ose (fixtures/provider.py:51)
2017-02-20 03:16:00,675 [E] assert <Thread(Thread-3, started daemon 140532765783808)> is None
 +  where <Thread(Thread-3, started daemon 140532765783808)> = <Thread(Thread-3, started daemon 140532765783808)>
 +    where <Thread(Thread-3, started daemon 140532765783808)> = <utils.browser.BrowserManager object at 0x3505610>._browser_renew_thread (fixtures/provider.py:52)
Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/cfme-5.7-ose-3.4-full-gate/cfme_tests/fixtures/provider.py", line 49, in _setup_provider
    providers.setup_provider(provider_key)
  File "/var/lib/jenkins/workspace/cfme-5.7-ose-3.4-full-gate/cfme_tests/utils/providers.py", line 284, in setup_provider
    check_existing=check_existing)
  File "/var/lib/jenkins/workspace/cfme-5.7-ose-3.4-full-gate/cfme_tests/cfme/containers/provider/openshift.py", line 31, in create
    super(OpenshiftProvider, self).create(validate_credentials=validate_credentials, **kwargs)
  File "/var/lib/jenkins/workspace/cfme-5.7-ose-3.4-full-gate/cfme_tests/cfme/common/provider.py", line 218, in create
    navigate_to(self, 'Add')
  File "/var/lib/jenkins/workspace/cfme-5.7-ose-3.4-full-gate/cfme_tests/.cfme_tests/lib/python2.7/site-packages/navmazing/__init__.py", line 104, in navigate
    return nav(cls_or_obj, self).go(0, *args, **kwargs)
  File "/var/lib/jenkins/workspace/cfme-5.7-ose-3.4-full-gate/cfme_tests/utils/appliance/implementations/ui.py", line 381, in go
    self.prerequisite_view = self.prerequisite()
  File "/var/lib/jenkins/workspace/cfme-5.7-ose-3.4-full-gate/cfme_tests/.cfme_tests/lib/python2.7/site-packages/navmazing/__init__.py", line 165, in __call__
    return self.obj.navigate_obj.navigate(self.obj.obj, self.target)
  File "/var/lib/jenkins/workspace/cfme-5.7-ose-3.4-full-gate/cfme_tests/.cfme_tests/lib/python2.7/site-packages/navmazing/__init__.py", line 104, in navigate
    return nav(cls_or_obj, self).go(0, *args, **kwargs)
  File "/var/lib/jenkins/workspace/cfme-5.7-ose-3.4-full-gate/cfme_tests/utils/appliance/implementations/ui.py", line 381, in go
    self.prerequisite_view = self.prerequisite()
  File "/var/lib/jenkins/workspace/cfme-5.7-ose-3.4-full-gate/cfme_tests/.cfme_tests/lib/python2.7/site-packages/navmazing/__init__.py", line 193, in __call__
    return self.obj.navigate_obj.navigate(attr, self.target)
  File "/var/lib/jenkins/workspace/cfme-5.7-ose-3.4-full-gate/cfme_tests/.cfme_tests/lib/python2.7/site-packages/navmazing/__init__.py", line 104, in navigate
    return nav(cls_or_obj, self).go(0, *args, **kwargs)
  File "/var/lib/jenkins/workspace/cfme-5.7-ose-3.4-full-gate/cfme_tests/utils/appliance/implementations/ui.py", line 381, in go
    self.prerequisite_view = self.prerequisite()
  File "/var/lib/jenkins/workspace/cfme-5.7-ose-3.4-full-gate/cfme_tests/.cfme_tests/lib/python2.7/site-packages/navmazing/__init__.py", line 165, in __call__
    return self.obj.navigate_obj.navigate(self.obj.obj, self.target)
  File "/var/lib/jenkins/workspace/cfme-5.7-ose-3.4-full-gate/cfme_tests/.cfme_tests/lib/python2.7/site-packages/navmazing/__init__.py", line 104, in navigate
    return nav(cls_or_obj, self).go(0, *args, **kwargs)
  File "/var/lib/jenkins/workspace/cfme-5.7-ose-3.4-full-gate/cfme_tests/utils/appliance/implementations/ui.py", line 381, in go
    self.prerequisite_view = self.prerequisite()
  File "/var/lib/jenkins/workspace/cfme-5.7-ose-3.4-full-gate/cfme_tests/cfme/base/ui.py", line 36, in prerequisite
    ensure_browser_open()
  File "/var/lib/jenkins/workspace/cfme-5.7-ose-3.4-full-gate/cfme_tests/utils/browser.py", line 402, in ensure_browser_open
    return manager.ensure_open(url_key=url_key)
  File "/var/lib/jenkins/workspace/cfme-5.7-ose-3.4-full-gate/cfme_tests/utils/browser.py", line 285, in ensure_open
    return self.start(url_key=url_key)
  File "/var/lib/jenkins/workspace/cfme-5.7-ose-3.4-full-gate/cfme_tests/utils/browser.py", line 328, in start
    self._browser_start_renew_thread()
  File "/var/lib/jenkins/workspace/cfme-5.7-ose-3.4-full-gate/cfme_tests/utils/browser.py", line 246, in _browser_start_renew_thread
    assert self._browser_renew_thread is None
AssertionError: assert <Thread(Thread-3, started daemon 140532765783808)> is None
 +  where <Thread(Thread-3, started daemon 140532765783808)> = <Thread(Thread-3, started daemon 140532765783808)>
 +    where <Thread(Thread-3, started daemon 140532765783808)> = <utils.browser.BrowserManager object at 0x3505610>._browser_renew_thread
```

My gut tels me that #4148 may be the cause, but  I didn't have time to investigate more.

**Comment by jaryn** [comment]
I can see also this on the tty sometimes:
```
08:17:19 ========================== 7 passed in 244.47 seconds ==========================
08:17:19 
                                                                                
collecting artifacts
08:17:19 
08:17:19 Exception KeyError: KeyError(139738501781312,) in <module 'threading' from '/usr/lib64/python2.7/threading.pyc'> ignored
08:17:19 Archiving artifacts
```

**Comment by mfalesni** [comment]
This error was resolved in #4249
